src/persist-client/src/lib.rs

```diff
@@ -639,6 +649,10 @@ impl PersistClient {
         T: Timestamp + Lattice + Codec64,
         D: Semigroup + Codec64 + Send + Sync,
     {
+        if !DANGEROUS_ENABLE_SCHEMA_EVOLUTION.get(&self.cfg.configs) {
+            panic!("tried to evolve the schema of a Persist shard without the feature enabled");
```

Comment:
We could return an error here instead of panicking, but because this should never get called in Prod I figured panicking was better. Happy to change though!

Reply:
Panic seems great to me